### PR TITLE
fix decision table result for hit policies with multiple entries

### PR DESCRIPTION
**Pull request by vzickner**
Behavior before:
  Execution a decision table with the hit policy rule order and output order resulted in either an array or the element, depending on how many rows are hit.
  The user of the Decision/DMN task now needs to lookup which of both exists and pick it accordingly.

Behavior after:
  For decision table with the hit policy rule order and output order the result is always returned as an array. Independent on how much entries are hit.

#### Check List:
* Unit tests: YES 
* Documentation: NO

### CODE REVIEW

File: modules/flowable-dmn-engine-configurator/src/test/java/org/flowable/dmn/test/runtime/DmnTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import org.flowable.engine.runtime.ProcessInstance;
+import org.flowable.engine.test.ConfigurationResource;
+import org.flowable.engine.test.Deployment;
+import org.flowable.engine.test.FlowableTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (dbmalkovsky) [inline modules/flowable-dmn-engine-configurator/src/test/java/org/flowable/dmn/test/runtime/DmnTaskTest.java:29]: FWIW, the ordering of the imports does not match project standards.

Comment (vzickner) [reply modules/flowable-dmn-engine-configurator/src/test/java/org/flowable/dmn/test/runtime/DmnTaskTest.java]: I changed the order of the imports

**Comment by tijsrademakers** [comment]
@vzickner, thanks for the PR. One thing we have to take care of is backward compatibility. Because this could lead to different runtime behaviour for running process and case instances. Can you maybe add an engine configuration option to enable or disable this new behaviour? The default could be to enable it.

**Comment by vzickner** [comment]
Hi @tijsrademakers, thank you! I added the property `alwaysUseArraysForDmnMultiHitPolicies` to the BPMN and CMMN engine

**Pull request merged**